tests/helper.py

```diff
@@ -25,6 +25,11 @@ def __init__(self):
         self.resin_exceptions = resin_exceptions
         self.base_request = BaseRequest()
         self.settings = Settings()
+
+        if 'api_endpoint' in self.credentials:
+            self.settings.set('api_endpoint', self.credentials['api_endpoint'])
+            self.settings.set('pine_endpoint', '%s/%s/' % (self.credentials['api_endpoint'], self.settings.get('api_version')))
```

Comment:
You can use '`{0}/{1}'.format(self.credentials['api_endpoint'], self.settings.get('api_version'))` here instead to ensure the consistency with other components.